cassandra/connection.py

```diff
@@ -436,9 +436,10 @@ def get_request_id(self):
         try:
             return self.request_ids.popleft()
         except IndexError:
-            self.highest_request_id += 1
+            new_request_id = self.highest_request_id + 1
             # in_flight checks should guarantee this
             assert self.highest_request_id <= self.max_request_id
```

Comment:
@mambocab I think the comparison here should be `new_request_id <= self.max_request_id`, otherwise `self.highest_request_id` will end up being `self.max_request_id + 1`